app/src/main/java/com/example/android/uamp/fragments/SettingsScreen.kt

```diff
@@ -0,0 +1,209 @@
+    navController: NavController
+) {
+    val mCheckedValue = remember { mutableStateOf(false) }
+    val audioManager =
+        ContextCompat.getSystemService(LocalContext.current, AudioManager::class.java)
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(color = colorResource(id = R.color.nowPlayingWhiteBackground))
+    ) {
+        TopAppBar(
+            title = { Text("Settings", style = MaterialTheme.typography.h5) },
+            navigationIcon = {
+                IconButton(onClick = { navController.navigate("nowplaying") }) {
+                    Icon(Icons.Filled.ArrowBack, contentDescription = null)
+                }
+            },
+            backgroundColor = Color.White
+        )
+        Column(modifier = Modifier.padding(start = 10.dp, end = 10.dp)) {
+            Row(
+                modifier = Modifier
+                    .padding(top = 20.dp)
+                    .fillMaxWidth(), verticalAlignment = Alignment.CenterVertically
+            ) {
+                Text(
+                    text = "Enable Spatial Audio", modifier = Modifier
+                        .weight(2f)
+                )
+                Switch(
+                    checked = mCheckedValue.value,
+                    onCheckedChange = {
+                        mCheckedValue.value = it
+                        toggleSpatialAudio(mainActivityViewModel, mCheckedValue.value)
+                    },
+                    colors = SwitchDefaults.colors(
+                        checkedTrackColor = colorResource(id = R.color.colorPrimaryDark),
+                        checkedThumbColor = Color.White,
+                        uncheckedTrackColor = Color.Gray
+                    ),
+                )
+            }
+            SpatialAudioOutput(audioManager = audioManager!!)
+        }
+    }
+}
+
+/**
+ * SpatialAudioOutput initializes a spatializer through which the current spatial audio characteristics
+ * of the current media item being played are displayed.
+ *
+ * @param audioManager to declare spatializer
+ */
+@Composable
+fun SpatialAudioOutput(audioManager: AudioManager) {
+
+    val spatializer = audioManager.spatializer
+
+    val attributes = AudioAttributes.Builder().setUsage(AudioAttributes.USAGE_MEDIA).setContentType(
+        C.AUDIO_CONTENT_TYPE_UNKNOWN
+    ).setAllowedCapturePolicy(C.ALLOW_CAPTURE_BY_ALL).build()
+    val audioFormat = AudioFormat.Builder()
+        .setEncoding(AudioFormat.ENCODING_PCM_16BIT)
+        .setChannelMask(AudioFormat.CHANNEL_OUT_5POINT1)
+        .build()
+
+    val canBeSpatialized = spatializer.canBeSpatialized(attributes, audioFormat)
+    val getImmersiveAudioLevel = spatializer.immersiveAudioLevel
+    val isEnabled = spatializer.isEnabled
+    val isAvailable = spatializer.isAvailable
+
+    // Introduced in API 33, be sure to use a compatible device
+    val isHeadTrackerAvailable = spatializer.isHeadTrackerAvailable
```

Comment:
As-is, the app will crash if it's run on a device that's not running Android 13+. We should only call `spatializer.isHeadTrackerAvailable` if the device is running Android 13 and above, and set `isHeadTrackerAvailable` to `false` if not. You can check this by seeing if `Build.VERSION.SDK_INT` is at least `Build.VERSION_CODES.TIRAMISU`.
https://developer.android.com/reference/android/os/Build.VERSION#SDK_INT